### PR TITLE
Add Django Rest Framework to fermentrack

### PR DESCRIPTION
**Pull request by deedos**
As regarded at #340 , I 'm working in a native app for controlling Fermentrack with reduced functionality (meaning not all actions can be done through the app). For that reason, I ended up using Django Rest Framework to create endpoints as documented at https://github.com/deedos/fermentrack/blob/master/app/api/drf/README.md . It adds a create_user functions and it 's token based authentication api.

It's my firts PR in a real project so I hope you understand some basic mistakes here or there. Anyway, a feedback is welcome and I can adjust urls, folders locations or anyelse you would like to.

Cheers and once again thanks for this great software

### CODE REVIEW

**Comment by deedos** [comment]
Hi @thorrak . Could you take a look at this PR ? thanks a lot

File: app/api/drf/views.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
+from django.contrib.auth.models import User
+from rest_framework import viewsets
+from rest_framework.decorators import api_view
+from rest_framework.permissions import IsAuthenticated, AllowAny
+from app.api.drf.serializers import CreateUserSerializer, UserSerializer, BeerSerializer, BrewPiDeviceSerializer, FermentationProfileSerializer, FermentationProfilePointSerializer, BeerLogPointSerializer
+from app.models import Beer, BrewPiDevice, FermentationProfile, FermentationProfilePoint, BeerLogPoint
+from rest_framework import generics
+
+class CreateUserViewSet(viewsets.ModelViewSet):
```
Comment (thorrak) [inline app/api/drf/views.py]: What's the thought with allowing users to be created without any authentication?

Although I expect the majority of Fermentrack builds to be behind a firewall, this seems pretty dangerous given that an exposed install of Fermentrack would be trivially hackable.

File: app/api/drf/views.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    http_method_names = ['post']
+
+    def post(self, request, format=None):
+        serializer = UserSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+class UserViewSet(viewsets.ModelViewSet):
```
Comment (thorrak) [inline app/api/drf/views.py]: Similarly, what's the thought with exposing the full user list to all users? How would this be used by an external app?

File: app/api/drf/views.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    API endpoint that allows users to be viewed or edited.
+    """
+    permission_classes = [IsAuthenticated,]
+    queryset = User.objects.all()
+    serializer_class = UserSerializer
+
+
+class BeerViewSet(viewsets.ModelViewSet):
+    """
+    API endpoint that allows users to be viewed or edited.
```
Comment (thorrak) [inline app/api/drf/views.py]: allows _beers_ to be viewed or edited - not users. ;)

File: requirements.txt [hunk 4]
```diff
@@ -28,3 +28,7 @@ aioblescan      # Replacement for beacontools for Tilt support
 pyzmq<17
 tornado<5
 
+# Django Rest Framework and related
+djangorestframework
+djangorestframework-jwt
```
Comment (thorrak) [inline requirements.txt]: I'll need to do some research on these two packages, as I haven't used them before. What does JWT do?

File: app/api/drf/README.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+    curl -X POST -H "Content-Type: application/json" -d '{"first_name": "Joe", "last_name":"Doe", "username": "joe", "password": "mypass", "email": "joedoe@riseup.net"}' http://your-url/api-v1/create_user/
+
+### Response Example
+
+    {"id":14,"username":"joe","first_name":"Joe","last_name":"Doe","email":"joedoe@riseup.net"}
+
+## Tokens (POST)
+
+This implementation uses  Django Rest Framework JWT Auth https://getblimp.github.io/django-rest-framework-jwt/ for authentications and token management. It also send user data along with token response.
```
Comment (thorrak) [inline app/api/drf/README.md]: This URL no longer seems valid, unfortunately

**Pull request closed without merging**